pkg/services/cosmosdb/common-provision.go

```diff
@@ -83,7 +83,70 @@ func (c *cosmosAccountManager) ValidateProvisioningParameters(
 			}
 		}
 	}
+	if pp.ConsistencyPolicy != nil {
+		cp := strings.ToLower(pp.ConsistencyPolicy.DefaultConsistency)
+		// treat this map like a set and use lookup to check the
+		// validuty of the given consistency policy
+		validValues := map[string]struct{}{
+			"eventual":         {},
+			"session":          {},
+			"boundedstaleness": {},
+			"strong":           {},
+			"consistentprefix": {},
+		}
```

Comment:
Good use for map[string]struct{}-- super efficient. Want to propose that this can be a _little_ bit moreso by defining the `validValues` at the `cosmosAccountManager ` level or even the package level so that we don't need to re-allocate memory for this static info on every invocation of this function.

Still not a blocker tho, imo.